OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -17,25 +18,37 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class Land : Activity
 	{
-		readonly Target target;
 		readonly Aircraft aircraft;
 		readonly WVec offset;
+		readonly int? desiredFacing;
 
+		Target target;
+		WPos targetPosition;
+		WPos w1, w2, w3;
+		CPos landingCell;
 		bool landingInitiated;
-		bool soundPlayed;
+		bool finishedApproach;
 
-		public Land(Actor self, Target t, WVec offset)
+		public Land(Actor self, Target? target = null, WVec? offset = null, int? facing = null)
```

Comment:
The problem with that is that the nullability of  `target` and `facing` does carry meaning. 

`target` needs to default to `null` because we have to distinguish between an invalid target and no target at all. In fact, I made a mistake here: When `target == null` the destination should be assigned on `OnFirstRun`, not in the constructor. Otherwise this wont work with queued orders. If we want to polish the landing location for fixed wing aircraft further in the future, then target would probably need to be assigned even later in the activity. If we assign `Target.Invalid` as a placeholder, we can no longer distinguish between the case when we should land in place and the case when we want to land on an actor that is dead or otherwise has become invalid.

`facing` needs to default to `null` as well, because when no facing is given we should not assume any magic value for facing. `aircraft.Info.InitialFacing` is only a valid default when `aircraft.Info.TurnToLand` is true, which isn't the case for any unit in the official mods as far as I'm aware. In the regular case a null facing should always stay null and be referenced throughout the `Land` activity to indicate no special turns need to be made.

I can work around this, but doing so would mean that any activity that calls `Land` as a childactivity and optionally passes a destination and/or facing (which is most of them now) would need to call `Land` multiple times in an if/else statement depending on whether the destination in the parent activity is null or not, etc. Any new activity which does so would need to have this as well leading to potential for errors and code bloat. Whereas if these parameters are nullable they can just be passed along directly (through multiple layers of parent/child activities if necessary) without regard for whether they are null or not.

Once again, I can work around these limitations if necessary (I'm working on it now), but it just doesn't sit well with me. If the idea is to not expose implementation details in the public interface, it seems that in a practical sense this would actually be doing the opposite.

If wrapping arguments in `Nullable<>` is really objectionable, we should probably look into adding support for NA values in our custom structs instead to handle cases such as these (there are probably many more cases where such a thing would come in handy). 

Thinking some more about it,  why is `Target` a `struct` in the first place and not a `class`?

Reply:
> If we assign `Target.Invalid` as a placeholder, we can no longer distinguish between the case when we should land in place and the case when we want to land on an actor that is dead or otherwise has become invalid.

Right. In this case, it makes sense to keep the target nullable internally, but then exposed via a constructor that takes a `Target` (which is assigned), and one that doesn't (which assigns `null`).

> `facing` needs to default to `null` as well, because when no facing is given we should not assume any magic value for facing. 

Our standard convention is to pass `facing: -1` in these cases.

> Thinking some more about it,  why is `Target` a `struct` in the first place and not a `class`?

Performance. Structs live on the stack or directly in the class memory on the heap, so avoid a level of pointer indirection to access.

Reply:
Replaced all `Nullable<>` wrappers in constructors. I also realized we already have NA values for 'Target' in a way, we just don't have a way to check for them without lumping in unrelated stuff like dead actors, so I added a simple check for that.